# HG changeset patch
# User Rik <rik@octave.org>
# Date 1386090249 28800
#      Tue Dec 03 09:04:09 2013 -0800
# Node ID 7c9148b235ebf52a6de9f8f1c2b5b4d6aef0f427
# Parent  e6c17e9bc60007e71a7a3c69cfef3e0781cfab2a
runtests.m: Also runtests for files that only contain %!xtest (bug #40780)

* runtests.m : Add xtest to regexp searching for test code in m-file.

diff --git a/scripts/testfun/runtests.m b/scripts/testfun/runtests.m
--- a/scripts/testfun/runtests.m
+++ b/scripts/testfun/runtests.m
@@ -122,17 +122,17 @@ function retval = has_functions (f)
   endif
 endfunction
 
 function retval = has_tests (f)
   fid = fopen (f);
   if (fid >= 0)
     str = fread (fid, "*char").';
     fclose (fid);
-    retval = ! isempty (regexp (str, '^%!(?:test|assert|error|warning)',
+    retval = ! isempty (regexp (str, '^%!(?:test|xtest|assert|error|warning)',
                                      'lineanchors', 'once'));
   else
     error ("runtests: fopen failed: %s", f);
   endif
 endfunction
 
 function print_pass_fail (n, p)
   if (n > 0)
